scipy/integrate/quadpack.py

```diff
@@ -750,11 +757,19 @@ def integrate(self, *args, **kwargs):
             f = self.func
         else:
             f = partial(self.integrate, depth=depth+1)
-
-        value, abserr = quad(f, low, high, args=args, **opt)
+        quad_r = quad(f, low, high, args=args, full_output=True, **opt)
```

Comment:
and a few lines lower, move ``infodict = quad_r[2]`` inside ``if self.full_output:``